Sam Schonsberg
1/16/2018
Python Quiz Prep

In [1]:
import numpy as np

First, define functions for $\alpha$, rule 3, and rule 4.

In [3]:
def alpha(L,dL,T_i,T_f):   #equation for coefficient of linear expansion
    dT = T_f - T_i
    a = dL / (L*dT)
    return a

def R3(delta_A,delta_B):  #applies to two variables being added/subtracted -- see page 8 of data treatment
    delta = np.sqrt(delta_A**2 + delta_B**2)
    return delta

def R4(Q, c, A,delta_A,exp_A, B,delta_B,exp_B, C,delta_C,exp_C):   #applies to three variable equations
    delta = np.abs(Q) * np.sqrt( (exp_A * (delta_A/A))**2 + (exp_B * (delta_B/B))**2 + (exp_C * (delta_C/C))**2)
    return delta

Now, define the variables we have.

In [4]:
L = 1.05 #m  length
unc_L = .001 #m   uncertainty in length

dL = 1.25 #m   change in length
unc_dL = 0.00001 #m   uncertainty in change in length

T_i = 23.4 #deg C
unc_T_i = 0.3 #deg C

T_f = 92.5 #deg C
unc_T_f = 0.1 #deg C

Finally, we plug these into our defined functions!

In [9]:
a = alpha(L,dL,T_i,T_f)

unc_dT = R3(unc_T_i,unc_T_f)

dT = T_f - T_i
unc_a = R4(a, 1, dL,1,unc_dL, L,unc_L,-1, dT,unc_dT,-1)

print 'The coefficient of linear expansion is '+str(round(a,5))+' +/- '+str(round(unc_a,5))+' 1/*C'

The coefficient of linear expansion is 0.01723 +/- 8e-05 1/*C


Nicely written with LaTe$\chi$:
$\alpha$ = 0.01723 $\pm$ 0.00008 ${C^\circ}^{-1}$